src/common/os_thread.h

```diff
@@ -79,9 +85,15 @@ typedef union {
 	char padding[512];
 } os_cpu_set_t;
 
+#ifdef __FreeBSD__
+#include <pthread.h>
```

Comment:
I predict some people are going to have a problem with including this here.
Why is it needed?
A lot of other headers are included all over the place anyways, so this wouldn't hurt a lot in NVML, but:
It would make sense to either not include pthread.h, or if it is included, then
```
typedef pthread_once_t os_once_t;
typedef pthread_cond_t os_cond_t;
etc...
```
I looks a bit strange to struggle with defining suitable opaque structs for linux and freebsd, and to include pthread.h a few lines later.